core/client/fs/src/main/java/alluxio/client/block/stream/LocalFilePacketReader.java

```diff
@@ -137,7 +137,10 @@ public Factory(FileSystemContext context, WorkerNetAddress address, long blockId
 
     @Override
     public PacketReader create(long offset, long len) throws IOException {
-      return new LocalFilePacketReader(mPath, offset, len, mPacketSize);
+      if (mReader == null) {
+        mReader = new LocalFileBlockReader(mPath);
```

Comment:
Thanks for creating this PR.
An issue with this change is there can only be one block reader per JVM, which would not work with concurrent reads.

Reply:
Sorry,I don't see that only be one block reader per JVM.You mean several blockInStreams share same factory?If so, this change may be wrong.But I don't think there is such situation,causes the factory has a close method without keep concurrent status.At this point I think it can change the file path to file reader.

Reply:
I see. This factory is created per `BlockInstream` creation. I think it's fine since this class is also annotated with `@NotThreadSafe`.

Also, can you share how much performance gain you see after applying this PR? This change caches `LocalFileBlockReader`. however, each call on `LocalFileBlockReader.read`still invokes `mLocalFileChannel.map`, which I thought is the bottleneck according to the JIRA description?

Reply:
The FileChannel.map is expensive operation too.I wrote a simple case to check the performance.In 10_000_000 bytes file, read one byte, then seek 10 bytes and so on.This PR get 8000 ms this is average value and master code is 20_000ms.There are 1_000_000 times open file in the case.So I checked that  just open and close file 1_000_000 times, and I found the time consumed equals the improvement after applying the PR. The Channel.map is bottleneck too.Currently the speed of invoking read and seek methods in turn is still slower than transferring all data to a cache and then seek it.

Reply:
Cool. Thanks for the information and it's good to know. Yes, the improvement observed from this PR is significant, and we can see how to improve the `map` overhead in future PRs.